meta-iotqa/lib/oeqa/runtime/peripherals/mraa/mraa_gpio.py

```diff
@@ -18,30 +44,89 @@ def setUp(self):
+        self.cmd_direction = '%s/direction' % self.cmd_direction
+        return (self.cmd_direction)
+
+    def check_gpio_value(self):
+        cmd = "cat /sys/class/gpio/gpio20/value".split()
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        return int(output)
+    def check_gpio_direction(self):
+        cmd = self.cmd_direction.split()
+        (status, output) = self.target.run('cat %s' % cmd)
+        output = str(output)
+        if 'out' in output:
+            output = 'out'
+            return output
+
+        elif 'in' in output:
+            output = 'in'
+            return output
+
+        else:
+            return output
+
+
+
+    def test_gpio_cmd_set(self):
         '''
         Test a GPIO pin on and off and check the pin output with
-        BeagleBone
+        BeagleBone. DUT GPIO Pin is shorted with BeagleBone to check value for mraa-gpio set
+        it is shorted to gpio pin numeber 20 in BeagleBone
         '''
-        def check_gpio_output():
-            cmd = "cat /sys/class/gpio/gpio20/value".split()
-            output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-            return int(output)
+
+        status, output = self.target.run("mraa-gpio set " + self.pin + " 1")
+        sleep(1)
+        gpio = self.check_gpio_value()
+        self.assertEqual(gpio, 1, msg="GPIO pin output is not 1, " +
+                                      "mraa output:\n" + output)
 
         status, output = self.target.run("mraa-gpio set " + self.pin + " 0")
         sleep(1)
-        gpio = check_gpio_output()
+        gpio = self.check_gpio_value()
         self.assertEqual(gpio, 0, msg="GPIO pin output is not 0, " +
                                       "mraa output:\n" + output)
 
         status, output = self.target.run("mraa-gpio set " + self.pin + " 1")
         sleep(1)
-        gpio = check_gpio_output()
+        gpio = self.check_gpio_value()
         self.assertEqual(gpio, 1, msg="GPIO pin output is not 1, " +
                                       "mraa output:\n" + output)
 
-        status, output = self.target.run("mraa-gpio set " + self.pin + " 0")
+    def test_gpio_cmd_list(self):
+
+        (status, output) = self.target.run("mraa-gpio list")
+        self.assertEqual(status, 0, msg="mraa-gpio list command failed: %s " % output)
+
+    def test_gpio_cmd_get(self):
+
+        output = self.check_gpio_pin_number()
+        print (output)
+
+        status, output = self.target.run("mraa-gpio set " + self.pin + " 1")
         sleep(1)
-        gpio = check_gpio_output()
-        self.assertEqual(gpio, 0, msg="GPIO pin output is not 0, " +
-                                      "mraa output:\n" + output)
+        gpio = self.check_gpio_direction()
+        self.assertEqual(gpio, "out", msg="GPIO pin direction is not out, " +
+                                          "mraa output:\n" + gpio)
+        status, output = self.target.run("mraa-gpio get " + self.pin)
```

Comment:
Even when I tested it manually by mra-gpio set first followed by mraa-gpio get on same board the value was not same but direction changes from out to in so I am not checking value , I am checking direction .
To check value for get I think we have to provide input from host board and the run get command and check for value .